### DISCUSSION TITLE
is a cpu seed different from a gpu seed

### DISCUSSION THREAD

**Discussion by Gracelesslemon** [mrdbourke/pytorch-deep-learning#971] [Q&A]
question 7 on the exercise in section 2
```
torch.manual_seed(1234)
torch.cuda.manual_seed(1234)
assert torch.cuda.is_available()
device="cuda" 
cpu_tensor=torch.rand(3,3)
gpu_tensor_a=cpu_tensor.to(device=device)
gpu_tensor_b=torch.rand(size=(3,3),device=device)
print(f"cpu tensor:{cpu_tensor}\n cpu tensor converted to gpu tensor:{gpu_tensor_a}\n gpu tensor:{gpu_tensor_b}")
print(f"are they equal :{torch.equal(gpu_tensor_a,gpu_tensor_b)}")
```

output
```
cpu tensor:tensor([[0.0290, 0.4019, 0.2598],
        [0.3666, 0.0583, 0.7006],
        [0.0518, 0.4681, 0.6738]])
 cpu tensor converted to gpu tensor:tensor([[0.0290, 0.4019, 0.2598],
        [0.3666, 0.0583, 0.7006],
        [0.0518, 0.4681, 0.6738]], device='cuda:0')
 gpu tensor:tensor([[0.1272, 0.8167, 0.5440],
        [0.6601, 0.2721, 0.9737],
        [0.3903, 0.3394, 0.5451]], device='cuda:0')
are they equal :False
```
should they be same .(it should be true no?)

**Comment by LuluW8071** [comment]
```py
import torch

# Set the seed for CPU operations
torch.manual_seed(42)

# Set the seed for CUDA operations
torch.cuda.manual_seed(42)
```

They should be different since
- `manual_seed()` affects random number generation in the system's RAM.
- `cuda.manual_seed()` affects random number generation in your system's VRAM _(not 100% sure about this one)_.

**Reply by Gracelesslemon** [reply]
ok cool they are supposed to be different .
the confirmation is all i wanted
Thanks
